qutebrowser/browser/commands.py

```diff
@@ -832,6 +833,60 @@ def paste(self, sel=False, tab=False, bg=False, window=False):
             self._open(url, tab, bg, window)
 
     @cmdutils.register(instance='command-dispatcher', scope='window',
+                       completion=[usertypes.Completion.tab])
+    def buffer(self, index):
+        """Select tab by index or url/title best match, focus window if
+        necessary.
+
+        Args:
+            index: The [win_id/]index of the tab to focus. Or a substring
+                   in which case the closest match will be focused.
+        """
+        index_parts = index.split('/', 1)
+
+        try:
+            for part in index_parts:
+                int(part)
+        except ValueError:
+            model = instances.get(usertypes.Completion.tab)
+            sf = sortfilter.CompletionFilterModel(source=model)
+            sf.set_pattern(index)
+            if sf.count() > 0:
+                index = sf.data(sf.first_item())
+                index_parts = index.split('/', 1)
+            else:
+                raise cmdexc.CommandError(
+                    "No matching tab for: {}".format(index))
+
+        if len(index_parts) == 2:
+            win_id = int(index_parts[0])
+            idx = int(index_parts[1])
+        elif len(index_parts) == 1:
+            idx = int(index_parts[0])
+            win_id = -1
+            active_win = objreg.get('app').activeWindow()
+            if active_win is None:
+                # Not sure how you enter a command without and active window...
+                raise cmdexc.CommandError(
+                    "No window specified and couldn't find active window!")
+            win_id = active_win.win_id
```

Comment:
Can you add an `else: raise AssertionError("Unexpected index length {}".format(len(index_parts)))` or so here, so there's a defined behaviour if that should ever happen (e.g. by someone changing the completion model in the future) rather than undefined behaviour?